tools/src/main/java/org/apache/kafka/trogdor/workload/ConsumeBenchWorker.java

```diff
@@ -86,48 +89,91 @@ public void start(Platform platform, WorkerStatusTracker status,
                     }
+                    partitionsByTopic.put(topic, partitions);
                 }
-                log.info("Will consume from {}", partitions);
-                executor.submit(new ConsumeMessages(partitions));
+
+                AbstractConsumeMessages consumeMessagesTask;
+                if (spec.consumerGroup() == null) {
+                    spec.consumerGroup(DEFAULT_CONSUMER_GROUP);
+                    consumeMessagesTask = new AssignConsumeMessages(partitionsByTopic);
+                } else
+                    consumeMessagesTask = new SubscribedConsumeMessage(partitionsByTopic);
+
+                log.info("Creating new task {}. Will consume from {}",
+                    consumeMessagesTask.getClass().getName(), partitionsByTopic);
+                executor.submit(consumeMessagesTask);
             } catch (Throwable e) {
                 WorkerUtils.abort(log, "Prepare", e, doneFuture);
             }
         }
     }
 
-    public class ConsumeMessages implements Callable<Void> {
+    public class SubscribedConsumeMessage extends AbstractConsumeMessages {
+        SubscribedConsumeMessage(Map<String, List<TopicPartition>> topicPartitionsByTopic) {
+            super(topicPartitionsByTopic);
+        }
+
+        @Override
+        KafkaConsumer<byte[], byte[]> createConsumer(Map<String, List<TopicPartition>> topicPartitionsByTopic) {
+            KafkaConsumer<byte[], byte[]> consumer = new KafkaConsumer<>(
+                this.consumerProperties, new ByteArrayDeserializer(), new ByteArrayDeserializer());
+            consumer.subscribe(topicPartitionsByTopic.keySet());
+            return consumer;
+        }
+    }
+
+    public class AssignConsumeMessages extends AbstractConsumeMessages {
+        AssignConsumeMessages(Map<String, List<TopicPartition>> topicPartitionsByTopic) {
+            super(topicPartitionsByTopic);
+        }
+
+        @Override
+        KafkaConsumer<byte[], byte[]> createConsumer(Map<String, List<TopicPartition>> topicPartitionsByTopic) {
+            List<TopicPartition> topicPartitions = topicPartitionsByTopic.values().stream()
+                .flatMap(List::stream).collect(Collectors.toList());
+            KafkaConsumer<byte[], byte[]> consumer = new KafkaConsumer<>(
+                consumerProperties, new ByteArrayDeserializer(), new ByteArrayDeserializer());
+            consumer.assign(topicPartitions);
+            return consumer;
+        }
+    }
+
+    public abstract class AbstractConsumeMessages implements Callable<Void> {
         private final Histogram latencyHistogram;
         private final Histogram messageSizeHistogram;
         private final Future<?> statusUpdaterFuture;
         private final Throttle throttle;
+        Properties consumerProperties;
 
-        ConsumeMessages(Collection<TopicPartition> topicPartitions) {
+        AbstractConsumeMessages(Map<String, List<TopicPartition>> topicPartitionsByTopic) {
```

Comment:
Seems like we don't really need inheritance here.  Can just have an "if" statement that checks if we have a group or not